From 8970c8040a87bb9da1d22cabfd5dafb79924ed1f Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 06:37:42 -0700
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/level/block/grower/TreeGrower.java b/net/minecraft/world/level/block/grower/TreeGrower.java
index ca4f7d31976cb55d0eaffb4af80b0cf2ddd2ad6d..d23f255de9208f42125fa358a9e8194c984fe4d3 100644
--- a/net/minecraft/world/level/block/grower/TreeGrower.java
+++ b/net/minecraft/world/level/block/grower/TreeGrower.java
@@ -132,6 +132,7 @@ public final class TreeGrower {
                 .get(configuredMegaFeature)
                 .orElse(null);
             if (holder != null) {
+                this.setTreeType(holder); // CraftBukkit
                 for (int i = 0; i >= -1; i--) {
                     for (int i1 = 0; i1 >= -1; i1--) {
                         if (isTwoByTwoSapling(state, level, pos, i, i1)) {
@@ -164,6 +165,7 @@ public final class TreeGrower {
             if (holder1 == null) {
                 return false;
             } else {
+                this.setTreeType(holder1); // CraftBukkit
                 ConfiguredFeature<?, ?> configuredFeature2 = holder1.value();
                 BlockState blockState1 = level.getFluidState(pos).createLegacyBlock();
                 level.setBlock(pos, blockState1, 260);
@@ -198,4 +200,58 @@ public final class TreeGrower {
 
         return false;
     }
+
+    // CraftBukkit start
+    private void setTreeType(Holder<ConfiguredFeature<?, ?>> feature) {
+        if (feature.is(TreeFeatures.OAK) || feature.is(TreeFeatures.OAK_BEES_005)) {
+            net.minecraft.world.level.block.SaplingBlock.treeType = org.bukkit.TreeType.TREE;
+        } else if (feature.is(TreeFeatures.HUGE_RED_MUSHROOM)) {
+            net.minecraft.world.level.block.SaplingBlock.treeType = org.bukkit.TreeType.RED_MUSHROOM;
+        } else if (feature.is(TreeFeatures.HUGE_BROWN_MUSHROOM)) {
+            net.minecraft.world.level.block.SaplingBlock.treeType = org.bukkit.TreeType.BROWN_MUSHROOM;
+        } else if (feature.is(TreeFeatures.JUNGLE_TREE)) {
+            net.minecraft.world.level.block.SaplingBlock.treeType = org.bukkit.TreeType.COCOA_TREE;
+        } else if (feature.is(TreeFeatures.JUNGLE_TREE_NO_VINE)) {
+            net.minecraft.world.level.block.SaplingBlock.treeType = org.bukkit.TreeType.SMALL_JUNGLE;
+        } else if (feature.is(TreeFeatures.PINE)) {
+            net.minecraft.world.level.block.SaplingBlock.treeType = org.bukkit.TreeType.TALL_REDWOOD;
+        } else if (feature.is(TreeFeatures.SPRUCE)) {
+            net.minecraft.world.level.block.SaplingBlock.treeType = org.bukkit.TreeType.REDWOOD;
+        } else if (feature.is(TreeFeatures.ACACIA)) {
+            net.minecraft.world.level.block.SaplingBlock.treeType = org.bukkit.TreeType.ACACIA;
+        } else if (feature.is(TreeFeatures.BIRCH) || feature.is(TreeFeatures.BIRCH_BEES_005)) {
+            net.minecraft.world.level.block.SaplingBlock.treeType = org.bukkit.TreeType.BIRCH;
+        } else if (feature.is(TreeFeatures.SUPER_BIRCH_BEES_0002)) {
+            net.minecraft.world.level.block.SaplingBlock.treeType = org.bukkit.TreeType.TALL_BIRCH;
+        } else if (feature.is(TreeFeatures.SWAMP_OAK)) {
+            net.minecraft.world.level.block.SaplingBlock.treeType = org.bukkit.TreeType.SWAMP;
+        } else if (feature.is(TreeFeatures.FANCY_OAK) || feature.is(TreeFeatures.FANCY_OAK_BEES_005)) {
+            net.minecraft.world.level.block.SaplingBlock.treeType = org.bukkit.TreeType.BIG_TREE;
+        } else if (feature.is(TreeFeatures.JUNGLE_BUSH)) {
+            net.minecraft.world.level.block.SaplingBlock.treeType = org.bukkit.TreeType.JUNGLE_BUSH;
+        } else if (feature.is(TreeFeatures.DARK_OAK)) {
+            net.minecraft.world.level.block.SaplingBlock.treeType = org.bukkit.TreeType.DARK_OAK;
+        } else if (feature.is(TreeFeatures.MEGA_SPRUCE)) {
+            net.minecraft.world.level.block.SaplingBlock.treeType = org.bukkit.TreeType.MEGA_REDWOOD;
+        } else if (feature.is(TreeFeatures.MEGA_PINE)) {
+            net.minecraft.world.level.block.SaplingBlock.treeType = org.bukkit.TreeType.MEGA_PINE;
+        } else if (feature.is(TreeFeatures.MEGA_JUNGLE_TREE)) {
+            net.minecraft.world.level.block.SaplingBlock.treeType = org.bukkit.TreeType.JUNGLE;
+        } else if (feature.is(TreeFeatures.AZALEA_TREE)) {
+            net.minecraft.world.level.block.SaplingBlock.treeType = org.bukkit.TreeType.AZALEA;
+        } else if (feature.is(TreeFeatures.MANGROVE)) {
+            net.minecraft.world.level.block.SaplingBlock.treeType = org.bukkit.TreeType.MANGROVE;
+        } else if (feature.is(TreeFeatures.TALL_MANGROVE)) {
+            net.minecraft.world.level.block.SaplingBlock.treeType = org.bukkit.TreeType.TALL_MANGROVE;
+        } else if (feature.is(TreeFeatures.CHERRY) || feature.is(TreeFeatures.CHERRY_BEES_005)) {
+            net.minecraft.world.level.block.SaplingBlock.treeType = org.bukkit.TreeType.CHERRY;
+        } else if (feature.is(TreeFeatures.PALE_OAK) || feature.is(TreeFeatures.PALE_OAK_BONEMEAL)) {
+            net.minecraft.world.level.block.SaplingBlock.treeType = org.bukkit.TreeType.PALE_OAK;
+        } else if (feature.is(TreeFeatures.PALE_OAK_CREAKING)) {
+            net.minecraft.world.level.block.SaplingBlock.treeType = org.bukkit.TreeType.PALE_OAK_CREAKING;
+        } else {
+            throw new IllegalArgumentException("Unknown tree generator " + feature);
+        }
+    }
+    // CraftBukkit end
 }
