From 8970c8040a87bb9da1d22cabfd5dafb79924ed1f Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 06:37:42 -0700
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/entity/projectile/windcharge/BreezeWindCharge.java b/net/minecraft/world/entity/projectile/windcharge/BreezeWindCharge.java
index f6dc0f8a52d597d8fe0e50f53a0a76bf6f96c36b..ea90fe3057b66b76d3dd0a155f2cdcd7bcd7e5bb 100644
--- a/net/minecraft/world/entity/projectile/windcharge/BreezeWindCharge.java
+++ b/net/minecraft/world/entity/projectile/windcharge/BreezeWindCharge.java
@@ -20,6 +20,12 @@ public class BreezeWindCharge extends AbstractWindCharge {
 
     @Override
     public void explode(Vec3 pos) {
+        // Paper start - Fire event for WindCharge explosions
+        org.bukkit.event.entity.ExplosionPrimeEvent event = org.bukkit.craftbukkit.event.CraftEventFactory.callExplosionPrimeEvent(this, RADIUS, false);
+        if (event.isCancelled()) {
+            return;
+        }
+        // Paper end - Fire event for WindCharge explosions
         this.level()
             .explode(
                 this,
@@ -28,8 +34,8 @@ public class BreezeWindCharge extends AbstractWindCharge {
                 pos.x(),
                 pos.y(),
                 pos.z(),
-                3.0F,
-                false,
+                event.getRadius(), // Paper - Fire event for WindCharge explosions
+                event.getFire(), // Paper - Fire event for WindCharge explosions
                 Level.ExplosionInteraction.TRIGGER,
                 ParticleTypes.GUST_EMITTER_SMALL,
                 ParticleTypes.GUST_EMITTER_LARGE,
