From 8970c8040a87bb9da1d22cabfd5dafb79924ed1f Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 06:37:42 -0700
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/entity/projectile/Arrow.java b/net/minecraft/world/entity/projectile/Arrow.java
index 7290d3b1cc87eaeca3d79b11544f3199cfa0305c..1f22f44abb21d1ed9a4870f668779efb8fd9b295 100644
--- a/net/minecraft/world/entity/projectile/Arrow.java
+++ b/net/minecraft/world/entity/projectile/Arrow.java
@@ -115,7 +115,7 @@ public class Arrow extends AbstractArrow {
         Entity effectSource = this.getEffectSource();
         PotionContents potionContents = this.getPotionContents();
         float potionDurationScale = this.getPotionDurationScale();
-        potionContents.forEachEffect(mobEffectInstance -> living.addEffect(mobEffectInstance, effectSource), potionDurationScale);
+        potionContents.forEachEffect(mobEffectInstance -> living.addEffect(mobEffectInstance, effectSource, org.bukkit.event.entity.EntityPotionEffectEvent.Cause.ARROW), potionDurationScale); // CraftBukkit
     }
 
     @Override
