From 8970c8040a87bb9da1d22cabfd5dafb79924ed1f Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 06:37:42 -0700
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/core/Rotations.java b/net/minecraft/core/Rotations.java
index 5c3920aadbdba1456da8a30a81b4d8e095567127..eeba78418f6b047f7712adf1254a6cfaf617ef6d 100644
--- a/net/minecraft/core/Rotations.java
+++ b/net/minecraft/core/Rotations.java
@@ -26,11 +26,22 @@ public record Rotations(float x, float y, float z) {
             buffer.writeFloat(value.z);
         }
     };
+    // Paper start - add internal method for skipping validation for plugins
+    private static boolean SKIP_VALIDATION = false;
+    public static Rotations createWithoutValidityChecks(float x, float y, float z) {
+        SKIP_VALIDATION = true;
+        Rotations rotations = new Rotations(x, y, z);
+        SKIP_VALIDATION = false;
+        return rotations;
+    }
+    // Paper end  - add internal method for skipping validation for plugins
 
     public Rotations(float x, float y, float z) {
+        if (SKIP_VALIDATION) { // Paper - add internal method for skipping validation for plugins
         x = !Float.isInfinite(x) && !Float.isNaN(x) ? x % 360.0F : 0.0F;
         y = !Float.isInfinite(y) && !Float.isNaN(y) ? y % 360.0F : 0.0F;
         z = !Float.isInfinite(z) && !Float.isNaN(z) ? z % 360.0F : 0.0F;
+        } // Paper - add internal method for skipping validation for plugins
         this.x = x;
         this.y = y;
         this.z = z;
