From 8970c8040a87bb9da1d22cabfd5dafb79924ed1f Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 06:37:42 -0700
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/network/HashedStack.java b/net/minecraft/network/HashedStack.java
index fa015aa502d229b3210b911fcb0dd8f3fe47c9bf..edc88c948e0eefc47835263fef76cb843682f346 100644
--- a/net/minecraft/network/HashedStack.java
+++ b/net/minecraft/network/HashedStack.java
@@ -17,7 +17,7 @@ public interface HashedStack {
         }
 
         @Override
-        public boolean matches(ItemStack stack, HashedPatchMap.HashGenerator hashGenerator) {
+        public boolean matches(ItemStack stack, HashedPatchMap.HashGenerator hashGenerator, final boolean simplifyMatching) { // Paper - add flag to simplify remote matching logic
             return stack.isEmpty();
         }
     };
@@ -27,7 +27,7 @@ public interface HashedStack {
             hashedStack -> hashedStack instanceof HashedStack.ActualItem actualItem ? Optional.of(actualItem) : Optional.empty()
         );
 
-    boolean matches(ItemStack stack, HashedPatchMap.HashGenerator hashGenerator);
+    boolean matches(ItemStack stack, HashedPatchMap.HashGenerator hashGenerator, final boolean simplifyMatching); // Paper - add flag to simplify remote matching logic
 
     static HashedStack create(ItemStack stack, HashedPatchMap.HashGenerator hashGenerator) {
         return (HashedStack)(stack.isEmpty()
@@ -47,10 +47,10 @@ public interface HashedStack {
         );
 
         @Override
-        public boolean matches(ItemStack stack, HashedPatchMap.HashGenerator hashGenerator) {
+        public boolean matches(ItemStack stack, HashedPatchMap.HashGenerator hashGenerator, final boolean simplifyMatching) { // Paper - add flag to simplify remote matching logic
             return this.count == stack.getCount()
                 && this.item.equals(stack.getItemHolder())
-                && this.components.matches(stack.getComponentsPatch(), hashGenerator);
+                && (simplifyMatching || this.components.matches(stack.getComponentsPatch(), hashGenerator)); // Paper - add flag to simplify remote matching logic
         }
     }
 }
