From 8970c8040a87bb9da1d22cabfd5dafb79924ed1f Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 06:37:42 -0700
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/effect/MobEffectInstance.java b/net/minecraft/world/effect/MobEffectInstance.java
index fa12ca4fc5560949a2cf9ef3e22d8eacf70b4d0a..0af74866c7dd58af2bcfde3d753416ff881cd8a4 100644
--- a/net/minecraft/world/effect/MobEffectInstance.java
+++ b/net/minecraft/world/effect/MobEffectInstance.java
@@ -219,7 +219,7 @@ public class MobEffectInstance implements Comparable<MobEffectInstance> {
             return false;
         } else {
             int i = this.isInfiniteDuration() ? entity.tickCount : this.duration;
-            if (this.effect.value().shouldApplyEffectTickThisTick(i, this.amplifier) && !this.effect.value().applyEffectTick(level, entity, this.amplifier)) {
+            if (this.effect.value().shouldApplyEffectTickThisTick(i, this.amplifier) && new io.papermc.paper.event.entity.EntityEffectTickEvent(entity.getBukkitLivingEntity(), org.bukkit.craftbukkit.potion.CraftPotionEffectType.minecraftHolderToBukkit(this.effect), this.amplifier).callEvent() && !this.effect.value().applyEffectTick(level, entity, this.amplifier)) { // Paper - Add EntityEffectTickEvent
                 return false;
             } else {
                 this.tickDownDuration();
@@ -415,7 +415,7 @@ public class MobEffectInstance implements Comparable<MobEffectInstance> {
                     .apply(instance, MobEffectInstance.Details::create)
             )
         );
-        public static final StreamCodec<ByteBuf, MobEffectInstance.Details> STREAM_CODEC = StreamCodec.recursive(
+        public static final StreamCodec<net.minecraft.network.FriendlyByteBuf, MobEffectInstance.Details> STREAM_CODEC = StreamCodec.recursive( // Paper - Track codec depth
             codec -> StreamCodec.composite(
                 ByteBufCodecs.VAR_INT,
                 MobEffectInstance.Details::amplifier,
@@ -427,7 +427,7 @@ public class MobEffectInstance implements Comparable<MobEffectInstance> {
                 MobEffectInstance.Details::showParticles,
                 ByteBufCodecs.BOOL,
                 MobEffectInstance.Details::showIcon,
-                codec.apply(ByteBufCodecs::optional),
+                codec.apply(ByteBufCodecs::increaseDepth).apply(ByteBufCodecs::optional), // Paper - Track codec depth
                 MobEffectInstance.Details::hiddenEffect,
                 MobEffectInstance.Details::new
             )
