From 8970c8040a87bb9da1d22cabfd5dafb79924ed1f Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 06:37:42 -0700
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/effect/OozingMobEffect.java b/net/minecraft/world/effect/OozingMobEffect.java
index f50a977150ec038ca73e887952ecb40480132505..2b02e9c58dd1abec0e8d2d2bc88332d4683aa309 100644
--- a/net/minecraft/world/effect/OozingMobEffect.java
+++ b/net/minecraft/world/effect/OozingMobEffect.java
@@ -49,7 +49,7 @@ class OozingMobEffect extends MobEffect {
         if (slime != null) {
             slime.setSize(2, true);
             slime.snapTo(x, y, z, level.getRandom().nextFloat() * 360.0F, 0.0F);
-            level.addFreshEntity(slime);
+            level.addFreshEntity(slime, org.bukkit.event.entity.CreatureSpawnEvent.SpawnReason.POTION_EFFECT); // CraftBukkit
         }
     }
 
