--- a/net/minecraft/world/food/FoodProperties.java
+++ b/net/minecraft/world/food/FoodProperties.java
@@ -56,7 +_,7 @@
                     Mth.randomBetween(random, 0.9F, 1.0F)
                 );
             } else {
-                player.getFoodData().eat(this);
+                player.getFoodData().eat(this, stack, (net.minecraft.server.level.ServerPlayer) player); // CraftBukkit
                 level.playSound(
                     null,
                     player.getX(),
