From 8970c8040a87bb9da1d22cabfd5dafb79924ed1f Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 06:37:42 -0700
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/entity/ConversionType.java b/net/minecraft/world/entity/ConversionType.java
index 7023ea73aac12e1366bd40858a3aa6324d0c8823..10898cdbdc42c9ccb046a10cd4305c007ed7a842 100644
--- a/net/minecraft/world/entity/ConversionType.java
+++ b/net/minecraft/world/entity/ConversionType.java
@@ -21,7 +21,7 @@ public enum ConversionType {
 
                 for (Entity entity : newMob.getPassengers()) {
                     entity.stopRiding();
-                    entity.remove(Entity.RemovalReason.DISCARDED);
+                    entity.remove(Entity.RemovalReason.DISCARDED, org.bukkit.event.entity.EntityRemoveEvent.Cause.TRANSFORMATION); // CraftBukkit - add Bukkit remove cause
                 }
 
                 firstPassenger.startRiding(newMob);
@@ -70,6 +70,7 @@ public enum ConversionType {
             if (leashHolder != null) {
                 oldMob.dropLeash();
             }
+            newMob.aware = oldMob.aware; // Paper - Fix nerfed slime when splitting
 
             this.convertCommon(oldMob, newMob, conversionParams);
         }
