From 8970c8040a87bb9da1d22cabfd5dafb79924ed1f Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 06:37:42 -0700
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/level/block/DirtPathBlock.java b/net/minecraft/world/level/block/DirtPathBlock.java
index 9e1cb22d8985010550ccf15e326d105801535fed..ed0aac42a17ed659f349f0888738b9feb10c6b51 100644
--- a/net/minecraft/world/level/block/DirtPathBlock.java
+++ b/net/minecraft/world/level/block/DirtPathBlock.java
@@ -60,6 +60,11 @@ public class DirtPathBlock extends Block {
 
     @Override
     protected void tick(BlockState state, ServerLevel level, BlockPos pos, RandomSource random) {
+        // CraftBukkit start - do not fade if the block is valid here
+        if (state.canSurvive(level, pos)) {
+            return;
+        }
+        // CraftBukkit end
         FarmBlock.turnToDirt(null, state, level, pos);
     }
 
