From 8970c8040a87bb9da1d22cabfd5dafb79924ed1f Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 06:37:42 -0700
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/item/trading/Merchant.java b/net/minecraft/world/item/trading/Merchant.java
index 4a5744fbac4fe742d3cbc7689fd5baf6c1fc57be..6e2fa442ddcd4ef3c4fe789b9ebac9dd409c9de7 100644
--- a/net/minecraft/world/item/trading/Merchant.java
+++ b/net/minecraft/world/item/trading/Merchant.java
@@ -19,6 +19,8 @@ public interface Merchant {
 
     void overrideOffers(MerchantOffers offers);
 
+    default void processTrade(MerchantOffer offer, @Nullable io.papermc.paper.event.player.PlayerPurchaseEvent event) { this.notifyTrade(offer); } // Paper
+
     void notifyTrade(MerchantOffer offer);
 
     void notifyTradeUpdated(ItemStack stack);
@@ -50,4 +52,6 @@ public interface Merchant {
     boolean isClientSide();
 
     boolean stillValid(Player player);
+
+    org.bukkit.craftbukkit.inventory.CraftMerchant getCraftMerchant(); // CraftBukkit
 }
