From 8970c8040a87bb9da1d22cabfd5dafb79924ed1f Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 06:37:42 -0700
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/inventory/AbstractCraftingMenu.java b/net/minecraft/world/inventory/AbstractCraftingMenu.java
index 5fb185c435d250ae265c13f358c962976d25cb13..a7953134164c8d4895889950afd6c4b18a5bd7cf 100644
--- a/net/minecraft/world/inventory/AbstractCraftingMenu.java
+++ b/net/minecraft/world/inventory/AbstractCraftingMenu.java
@@ -12,14 +12,17 @@ import net.minecraft.world.item.crafting.RecipeHolder;
 public abstract class AbstractCraftingMenu extends RecipeBookMenu {
     private final int width;
     private final int height;
-    public final CraftingContainer craftSlots;
+    public final TransientCraftingContainer craftSlots; // CraftBukkit
     public final ResultContainer resultSlots = new ResultContainer();
 
-    public AbstractCraftingMenu(MenuType<?> menuType, int containerId, int width, int height) {
+    public AbstractCraftingMenu(MenuType<?> menuType, int containerId, int width, int height, Inventory playerInventory) { // CraftBukkit
         super(menuType, containerId);
         this.width = width;
         this.height = height;
-        this.craftSlots = new TransientCraftingContainer(this, width, height);
+        // CraftBukkit start
+        this.craftSlots = new TransientCraftingContainer(this, width, height, playerInventory.player);
+        this.craftSlots.resultInventory = this.resultSlots; // let InventoryCrafting know about its result slot
+        // CraftBukkit end
     }
 
     protected Slot addResultSlot(Player player, int x, int y) {
