From 8970c8040a87bb9da1d22cabfd5dafb79924ed1f Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 06:37:42 -0700
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/inventory/HorseInventoryMenu.java b/net/minecraft/world/inventory/HorseInventoryMenu.java
index 521e241f1b379dbc625d3d2255e1d05cd492f319..6638b715dab2f255ae6da1620d175bd15f441e93 100644
--- a/net/minecraft/world/inventory/HorseInventoryMenu.java
+++ b/net/minecraft/world/inventory/HorseInventoryMenu.java
@@ -19,9 +19,23 @@ public class HorseInventoryMenu extends AbstractContainerMenu {
     public static final int SLOT_SADDLE = 0;
     public static final int SLOT_BODY_ARMOR = 1;
     public static final int SLOT_HORSE_INVENTORY_START = 2;
+    // CraftBukkit start
+    private @javax.annotation.Nullable org.bukkit.craftbukkit.inventory.CraftInventoryView view;
+    private final Inventory inventory;
+
+    @Override
+    public org.bukkit.inventory.InventoryView getBukkitView() {
+        if (this.view != null) {
+            return this.view;
+        }
+
+        return this.view = new org.bukkit.craftbukkit.inventory.CraftInventoryView(this.inventory.player.getBukkitEntity(), this.horseContainer.getOwner().getInventory(), this);
+    }
+    // CraftBukkit end
 
     public HorseInventoryMenu(int containerId, Inventory inventory, Container horseContainer, final AbstractHorse horse, int columns) {
         super(null, containerId);
+        this.inventory = inventory; // CraftBukkit
         this.horseContainer = horseContainer;
         this.horse = horse;
         horseContainer.startOpen(inventory.player);
