From 8970c8040a87bb9da1d22cabfd5dafb79924ed1f Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 06:37:42 -0700
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/inventory/PlayerEnderChestContainer.java b/net/minecraft/world/inventory/PlayerEnderChestContainer.java
index 22663325d7377a904b5c6e100715f09c09f05dce..bc2b95973192069fc64581b59583b19df876f55d 100644
--- a/net/minecraft/world/inventory/PlayerEnderChestContainer.java
+++ b/net/minecraft/world/inventory/PlayerEnderChestContainer.java
@@ -12,9 +12,22 @@ import net.minecraft.world.level.block.entity.EnderChestBlockEntity;
 public class PlayerEnderChestContainer extends SimpleContainer {
     @Nullable
     private EnderChestBlockEntity activeChest;
+    // CraftBukkit start
+    private final Player owner;
 
-    public PlayerEnderChestContainer() {
+    public org.bukkit.inventory.InventoryHolder getBukkitOwner() {
+        return this.owner.getBukkitEntity();
+    }
+
+    @Override
+    public @Nullable org.bukkit.Location getLocation() {
+        return this.activeChest != null ? org.bukkit.craftbukkit.util.CraftLocation.toBukkit(this.activeChest.getBlockPos(), this.activeChest.getLevel().getWorld()) : null;
+    }
+
+    public PlayerEnderChestContainer(Player owner) {
         super(27);
+        this.owner = owner;
+        // CraftBukkit end
     }
 
     public void setActiveChest(EnderChestBlockEntity enderChestBlockEntity) {
