From 8970c8040a87bb9da1d22cabfd5dafb79924ed1f Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 06:37:42 -0700
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/inventory/RemoteSlot.java b/net/minecraft/world/inventory/RemoteSlot.java
index 8267235c43e7cc605a7a7edc18212b8a25163b82..11bb98c92f901d9f55e147fefb336dee5fe5c2df 100644
--- a/net/minecraft/world/inventory/RemoteSlot.java
+++ b/net/minecraft/world/inventory/RemoteSlot.java
@@ -29,12 +29,14 @@ public interface RemoteSlot {
 
     public static class Synchronized implements RemoteSlot {
         private final HashedPatchMap.HashGenerator hasher;
+        private final boolean simplifyMatching; // Paper - add flag to simplify remote matching logic
         @Nullable
         private ItemStack remoteStack = null;
         @Nullable
         private HashedStack remoteHash = null;
 
-        public Synchronized(HashedPatchMap.HashGenerator hasher) {
+        public Synchronized(HashedPatchMap.HashGenerator hasher, final boolean simplifyMatching) { // Paper - add flag to simplify remote matching logic
+            this.simplifyMatching = simplifyMatching; // Paper - add flag to simplify remote matching logic
             this.hasher = hasher;
         }
 
@@ -54,7 +56,7 @@ public interface RemoteSlot {
         public boolean matches(ItemStack stack) {
             if (this.remoteStack != null) {
                 return ItemStack.matches(this.remoteStack, stack);
-            } else if (this.remoteHash != null && this.remoteHash.matches(stack, this.hasher)) {
+            } else if (this.remoteHash != null && this.remoteHash.matches(stack, this.hasher, this.simplifyMatching)) { // Paper - add flag to simplify remote matching logic
                 this.remoteStack = stack.copy();
                 return true;
             } else {
