From 8970c8040a87bb9da1d22cabfd5dafb79924ed1f Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 06:37:42 -0700
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/inventory/ResultSlot.java b/net/minecraft/world/inventory/ResultSlot.java
index e19f3597110bf5e3830c056f54ed91059d525a59..e4cba45c327d96550a92cd5f9a30b1e5bd212747 100644
--- a/net/minecraft/world/inventory/ResultSlot.java
+++ b/net/minecraft/world/inventory/ResultSlot.java
@@ -72,7 +72,7 @@ public class ResultSlot extends Slot {
     private NonNullList<ItemStack> getRemainingItems(CraftingInput input, Level level) {
         return level instanceof ServerLevel serverLevel
             ? serverLevel.recipeAccess()
-                .getRecipeFor(RecipeType.CRAFTING, input, serverLevel)
+                .getRecipeFor(RecipeType.CRAFTING, input, serverLevel, this.craftSlots.getCurrentRecipe()) // Paper - Perf: Improve mass crafting; check last recipe used first
                 .map(recipe -> recipe.value().getRemainingItems(input))
                 .orElseGet(() -> copyAllInputItems(input))
             : CraftingRecipe.defaultCraftingReminder(input);
