From 8970c8040a87bb9da1d22cabfd5dafb79924ed1f Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 06:37:42 -0700
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/item/component/WrittenBookContent.java b/net/minecraft/world/item/component/WrittenBookContent.java
index 540c2a14a0fa955db4e18c7d3d5fb897125231fe..eedf171b47c972920afdf8e25fb94e261ee65e13 100644
--- a/net/minecraft/world/item/component/WrittenBookContent.java
+++ b/net/minecraft/world/item/component/WrittenBookContent.java
@@ -94,7 +94,7 @@ public record WrittenBookContent(Filterable<String> title, String author, int ge
 
     public static boolean resolveForItem(ItemStack stack, CommandSourceStack source, @Nullable Player player) {
         WrittenBookContent writtenBookContent = stack.get(DataComponents.WRITTEN_BOOK_CONTENT);
-        if (writtenBookContent != null && !writtenBookContent.resolved()) {
+        if (io.papermc.paper.configuration.GlobalConfiguration.get().itemValidation.resolveSelectorsInBooks && writtenBookContent != null && !writtenBookContent.resolved()) { // Paper - Disable component selector resolving in books by default
             WrittenBookContent writtenBookContent1 = writtenBookContent.resolve(source, player);
             if (writtenBookContent1 != null) {
                 stack.set(DataComponents.WRITTEN_BOOK_CONTENT, writtenBookContent1);
