From 8970c8040a87bb9da1d22cabfd5dafb79924ed1f Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 06:37:42 -0700
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/network/protocol/game/ClientboundSetEquipmentPacket.java b/net/minecraft/network/protocol/game/ClientboundSetEquipmentPacket.java
index 7810c637c707dd7678461f2b554a7971cd60b92f..13f5fc753558a8a3d7e0075a3b8dd8178b5fd3c2 100644
--- a/net/minecraft/network/protocol/game/ClientboundSetEquipmentPacket.java
+++ b/net/minecraft/network/protocol/game/ClientboundSetEquipmentPacket.java
@@ -19,6 +19,13 @@ public class ClientboundSetEquipmentPacket implements Packet<ClientGamePacketLis
     private final List<Pair<EquipmentSlot, ItemStack>> slots;
 
     public ClientboundSetEquipmentPacket(int entity, List<Pair<EquipmentSlot, ItemStack>> slots) {
+    // Paper start - data sanitization
+        this(entity, slots, false);
+    }
+    private boolean sanitize;
+    public ClientboundSetEquipmentPacket(int entity, List<Pair<EquipmentSlot, ItemStack>> slots, boolean sanitize) {
+        this.sanitize = sanitize;
+    // Paper end - data sanitization
         this.entity = entity;
         this.slots = slots;
     }
@@ -40,6 +47,7 @@ public class ClientboundSetEquipmentPacket implements Packet<ClientGamePacketLis
         buffer.writeVarInt(this.entity);
         int size = this.slots.size();
 
+        try (final io.papermc.paper.util.sanitizer.ItemObfuscationSession ignored = io.papermc.paper.util.sanitizer.ItemObfuscationSession.start(this.sanitize ? io.papermc.paper.configuration.GlobalConfiguration.get().anticheat.obfuscation.items.binding.level : io.papermc.paper.util.sanitizer.ItemObfuscationSession.ObfuscationLevel.NONE)) { // Paper - data sanitization
         for (int i = 0; i < size; i++) {
             Pair<EquipmentSlot, ItemStack> pair = this.slots.get(i);
             EquipmentSlot equipmentSlot = pair.getFirst();
@@ -48,6 +56,7 @@ public class ClientboundSetEquipmentPacket implements Packet<ClientGamePacketLis
             buffer.writeByte(flag ? ordinal | -128 : ordinal);
             ItemStack.OPTIONAL_STREAM_CODEC.encode(buffer, pair.getSecond());
         }
+        } // Paper - data sanitization
     }
 
     @Override
