From 8970c8040a87bb9da1d22cabfd5dafb79924ed1f Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 06:37:42 -0700
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/network/protocol/game/ServerboundSetCreativeModeSlotPacket.java b/net/minecraft/network/protocol/game/ServerboundSetCreativeModeSlotPacket.java
index aa5b353856d715c90e077d1dcfaf0a0c8a92cf8a..83a5f53d4dfe27a40ad510037106b61a02c938ad 100644
--- a/net/minecraft/network/protocol/game/ServerboundSetCreativeModeSlotPacket.java
+++ b/net/minecraft/network/protocol/game/ServerboundSetCreativeModeSlotPacket.java
@@ -11,7 +11,7 @@ public record ServerboundSetCreativeModeSlotPacket(short slotNum, ItemStack item
     public static final StreamCodec<RegistryFriendlyByteBuf, ServerboundSetCreativeModeSlotPacket> STREAM_CODEC = StreamCodec.composite(
         ByteBufCodecs.SHORT,
         ServerboundSetCreativeModeSlotPacket::slotNum,
-        ItemStack.validatedStreamCodec(ItemStack.OPTIONAL_UNTRUSTED_STREAM_CODEC),
+        ItemStack.validatedStreamCodec(ItemStack.OPTIONAL_UNTRUSTED_STREAM_CODEC).apply(ByteBufCodecs::trackDepth), // Paper - Track codec depth
         ServerboundSetCreativeModeSlotPacket::itemStack,
         ServerboundSetCreativeModeSlotPacket::new
     );
