From 8970c8040a87bb9da1d22cabfd5dafb79924ed1f Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 06:37:42 -0700
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/server/commands/RideCommand.java b/net/minecraft/server/commands/RideCommand.java
index c3d3050e6cf8cbc9c2b19b7c6f40d0d9e5035363..281c54b870f095743f0d32180070b41bf85440f2 100644
--- a/net/minecraft/server/commands/RideCommand.java
+++ b/net/minecraft/server/commands/RideCommand.java
@@ -58,7 +58,7 @@ public class RideCommand {
         Entity vehicle1 = target.getVehicle();
         if (vehicle1 != null) {
             throw ERROR_ALREADY_RIDING.create(target.getDisplayName(), vehicle1.getDisplayName());
-        } else if (vehicle.getType() == EntityType.PLAYER) {
+        } else if (vehicle.getType() == EntityType.PLAYER && !io.papermc.paper.configuration.GlobalConfiguration.get().commands.rideCommandAllowPlayerAsVehicle) { // Paper - allow player as vehicle
             throw ERROR_MOUNTING_PLAYER.create();
         } else if (target.getSelfAndPassengers().anyMatch(passenger -> passenger == vehicle)) {
             throw ERROR_MOUNTING_LOOP.create();
