From 8970c8040a87bb9da1d22cabfd5dafb79924ed1f Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 06:37:42 -0700
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/entity/vehicle/MinecartTNT.java b/net/minecraft/world/entity/vehicle/MinecartTNT.java
index 565f89a630f15780bb0b1e8291b3000a51483111..70fcb6142880d7bacddfa0856c8c919304129367 100644
--- a/net/minecraft/world/entity/vehicle/MinecartTNT.java
+++ b/net/minecraft/world/entity/vehicle/MinecartTNT.java
@@ -38,6 +38,7 @@ public class MinecartTNT extends AbstractMinecart {
     public int fuse = -1;
     public float explosionPowerBase = 4.0F;
     public float explosionSpeedFactor = 1.0F;
+    public boolean isIncendiary = false; // CraftBukkit
 
     public MinecartTNT(EntityType<? extends MinecartTNT> entityType, Level level) {
         super(entityType, level);
@@ -52,6 +53,12 @@ public class MinecartTNT extends AbstractMinecart {
     public void tick() {
         super.tick();
         if (this.fuse > 0) {
+            // Paper start - Configurable TNT height nerf
+            if (this.level().paperConfig().fixes.tntEntityHeightNerf.test(v -> this.getY() > v)) {
+                this.discard(org.bukkit.event.entity.EntityRemoveEvent.Cause.OUT_OF_WORLD);
+                return;
+            }
+            // Paper end - Configurable TNT height nerf
             this.fuse--;
             this.level().addParticle(ParticleTypes.SMOKE, this.getX(), this.getY() + 0.5, this.getZ(), 0.0, 0.0, 0.0);
         } else if (this.fuse == 0) {
@@ -107,6 +114,17 @@ public class MinecartTNT extends AbstractMinecart {
         if (this.level() instanceof ServerLevel serverLevel) {
             if (serverLevel.getGameRules().getBoolean(GameRules.RULE_TNT_EXPLODES)) {
                 double min = Math.min(Math.sqrt(radiusModifier), 5.0);
+                // CraftBukkit start
+                org.bukkit.event.entity.ExplosionPrimeEvent event = new org.bukkit.event.entity.ExplosionPrimeEvent(
+                    this.getBukkitEntity(),
+                    (float) (this.explosionPowerBase + this.explosionSpeedFactor * this.random.nextDouble() * 1.5 * min),
+                    this.isIncendiary
+                );
+                if (!event.callEvent()) {
+                    this.fuse = -1;
+                    return;
+                }
+                // CraftBukkit end
                 serverLevel.explode(
                     this,
                     damageSource,
@@ -114,13 +132,13 @@ public class MinecartTNT extends AbstractMinecart {
                     this.getX(),
                     this.getY(),
                     this.getZ(),
-                    (float)(this.explosionPowerBase + this.explosionSpeedFactor * this.random.nextDouble() * 1.5 * min),
-                    false,
+                    event.getRadius(), // CraftBukkit - explosion prime event
+                    event.getFire(), // CraftBukkit - explosion prime event
                     Level.ExplosionInteraction.TNT
                 );
-                this.discard();
+                this.discard(org.bukkit.event.entity.EntityRemoveEvent.Cause.EXPLODE); // CraftBukkit - add Bukkit remove cause
             } else if (this.isPrimed()) {
-                this.discard();
+                this.discard(null); // CraftBukkit
             }
         }
     }
