From 8970c8040a87bb9da1d22cabfd5dafb79924ed1f Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 06:37:42 -0700
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/item/MaceItem.java b/net/minecraft/world/item/MaceItem.java
index c5fa2d1a3d3a362032df1fc023460e9188feeee5..a0bc395fa6700da397262e4b722593a915975200 100644
--- a/net/minecraft/world/item/MaceItem.java
+++ b/net/minecraft/world/item/MaceItem.java
@@ -52,7 +52,13 @@ public class MaceItem extends Item {
 
     @Override
     public void hurtEnemy(ItemStack stack, LivingEntity target, LivingEntity attacker) {
-        if (canSmashAttack(attacker)) {
+        // Paper start - Add EntityAttemptSmashAttackEvent
+        final boolean canSmashAttack = canSmashAttack(attacker);
+        io.papermc.paper.event.entity.EntityAttemptSmashAttackEvent event = new io.papermc.paper.event.entity.EntityAttemptSmashAttackEvent(attacker.getBukkitLivingEntity(), target.getBukkitLivingEntity(), stack.asBukkitCopy(), canSmashAttack);
+        event.callEvent();
+        final org.bukkit.event.Event.Result result = event.getResult();
+        if (result == org.bukkit.event.Event.Result.ALLOW || (canSmashAttack && result == org.bukkit.event.Event.Result.DEFAULT)) {
+        // Paper end - Add EntityAttemptSmashAttackEvent
             ServerLevel serverLevel = (ServerLevel)attacker.level();
             attacker.setDeltaMovement(attacker.getDeltaMovement().with(Direction.Axis.Y, 0.01F));
             if (attacker instanceof ServerPlayer serverPlayer) {
@@ -127,7 +133,7 @@ public class MaceItem extends Item {
             double knockbackPower = getKnockbackPower(attacker, livingEntity, vec3);
             Vec3 vec31 = vec3.normalize().scale(knockbackPower);
             if (knockbackPower > 0.0) {
-                livingEntity.push(vec31.x, 0.7F, vec31.z);
+                livingEntity.push(vec31.x, 0.7F, vec31.z, attacker); // Paper - Add EntityKnockbackByEntityEvent and EntityPushedByEntityAttackEvent
                 if (livingEntity instanceof ServerPlayer serverPlayer) {
                     serverPlayer.connection.send(new ClientboundSetEntityMotionPacket(serverPlayer));
                 }
