From 8970c8040a87bb9da1d22cabfd5dafb79924ed1f Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 06:37:42 -0700
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/entity/monster/Monster.java b/net/minecraft/world/entity/monster/Monster.java
index be49f56bed8b4384d66f1f1fe866ebf7060ee80e..5ef9566b16a4d0300ee45a993c46e734db156416 100644
--- a/net/minecraft/world/entity/monster/Monster.java
+++ b/net/minecraft/world/entity/monster/Monster.java
@@ -92,7 +92,7 @@ public abstract class Monster extends PathfinderMob implements Enemy {
             return false;
         } else {
             DimensionType dimensionType = level.dimensionType();
-            int i = dimensionType.monsterSpawnBlockLightLimit();
+            int i = level.getLevel().paperConfig().entities.spawning.monsterSpawnMaxLightLevel.or(dimensionType.monsterSpawnBlockLightLimit()); // Paper - Configurable max block light for monster spawning
             if (i < 15 && level.getBrightness(LightLayer.BLOCK, pos) > i) {
                 return false;
             } else {
